src/NUnitFramework/framework/Internal/Execution/WorkItem.cs

```diff
@@ -488,12 +490,8 @@ private ParallelExecutionStrategy GetExecutionStrategy()
                 Test is TestFixture && Context.ParallelScope.HasFlag(ParallelScope.Fixtures))
                 return ParallelExecutionStrategy.Parallel;
 
-            // There is no scope specified either on the item itself or in the context.
-            // In that case, simple work items are test cases and just run on the same
-            // thread, while composite work items and teardowns are non-parallel.
-            return this is SimpleWorkItem
-                ? ParallelExecutionStrategy.Direct
-                : ParallelExecutionStrategy.NonParallel;
+            // If all else fails, run on same thread
+            return ParallelExecutionStrategy.Direct;
```

Comment:
Let's say the child item represents a test fixture. If it has no attribute, and there is no context setting that says we should run fixtures in parallel, then it's supposed to run in isolation from all other tests. If we are currently on a parallel worker thread and run that test fixture directly, then it will run in parallel with other threads. That's the reason we need to use the nonparallel strategy for composite items.